Add CITE post-processing to docs (#110)
AlexsLemonade/scpca-docs: docs/sce_file_contents.md

Reviewer: Update needed on line 57 to include how the `scpca_filter` may also be based on CITE-seq data?

| `filtering_method`  | The method used for cell filtering. One of `emptyDrops`, `emptyDropsCellRanger`, or `UMI cutoff`. Only present for `filtered` objects |
| `umi_cutoff`        | The minimum UMI count per cell used as a threshold for removing empty droplets. Only present for `filtered` objects where the `filtering_method` is `UMI cutoff` |
| `prob_compromised_cutoff`        | The minimum cutoff for the probability of a cell being compromised, as calculated by `miQC`. Only present for `filtered` objects |

> | `scpca_filter_method`        | Method used by the Data Lab to filter low quality cells prior to normalization. Either `miQC` or `Minimum_gene_cutoff`; If CITE-seq data is also present, this will also indicate the ADT filtering method as `cleanTagCounts`, e.g. `miQC;cleanTagCounts`. Only present for `filtered` objects.   |

Reviewer: Either a period or lower case "if"
```suggestion
| `scpca_filter_method`        | Method used by the Data Lab to filter low quality cells prior to normalization. Either `miQC` or `Minimum_gene_cutoff`. If CITE-seq data is also present, this will also indicate the ADT filtering method as `cleanTagCounts`, e.g. `miQC;cleanTagCounts`. Only present for `filtered` objects.   |
```

| `altexps_CITEseq_sum`      | UMI count for CITE-seq ADTs                       |
| `altexps_CITEseq_detected` | Number of ADTs detected per cell (ADT count > 0 ) |
| `altexps_CITEseq_percent`  | Percent of `total` UMI count from ADT reads       |

In addition, the following QC statistics from [`DropletUtils::cleanTagCounts()`](https://rdrr.io/github/MarioniLab/DropletUtils/man/cleanTagCounts.html) can be found in the `colData` of the `"CITEseq"` alternative experiment, accessed with `colData(altExp(sce, "CITEseq"))`.

| Column name                | Contents                                          |
| -------------------------- | ------------------------------------------------- |
| `zero.ambient`   | Indicates whether the cell has zero ambient contamination   |
| `sum.controls` |  The sum of counts for all control features. Only present if negative control ADTs are present. |
| `high.controls`  | Indicates whether the cell has unusually high total control counts. Only present if negative control ADTs are present.|
| `ambient.scale` |  The relative amount of ambient contamination. Only present if negative control ADTs are _not_ present. |

> | `high.ambient`  | Indicates whether the cell has unusually high contamination. Only present if negative control ADTs are _not_ present.|

Reviewer: I just noticed this, and suddenly I don't know how I feel about it! (Not anything I would do differently at our end!)

Author: Well I can guarantee you a fun "wait, why are people doing what they are doing?" discussion next Tuesday at journal club to feel all these feelings all over again!

> | `discard`  | Indicates whether the cell should be discarded based on QC statistics. All retained cells in the `_processed.rds` are `FALSE`. |

Reviewer: I think I'm leaning toward removal. 

```suggestion
| `discard`  | Indicates whether the cell should be discarded based on QC statistics. |
```

altExp(sce, "CITEseq")
```

> Within this, the main expression matrix is again found in the `counts` assay and the normalized expression matrix is again found in the `logcounts` assay.

Reviewer: I don't feel like we need the double again
```suggestion
Within this, the main expression matrix is again found in the `counts` assay and the normalized expression matrix is found in the `logcounts` assay.
```

For each assay, each column corresponds to a cell or droplet (in the same order as the parent `SingleCellExperiment`) and each row corresponds to an antibody derived tag (ADT).
Column names are again cell barcode sequences and row names are the antibody targets for each ADT.

The following additional per-cell data columns for the CITE-seq data can be found in the main `colData` data frame (accessed with `colData(sce)` [as above](#cell-metrics)).

| Column name                | Contents                                          |
| -------------------------- | ------------------------------------------------- |

> | `altexps_CITEseq_sum`    | UMI count for CITE-seq ADTs                       |

Reviewer: As long as we had spacing here...
```suggestion
| `altexps_CITEseq_sum`      | UMI count for CITE-seq ADTs                       |
```

| `altexps_CITEseq_detected` | Number of ADTs detected per cell (ADT count > 0 ) |
| `altexps_CITEseq_percent`  | Percent of `total` UMI count from ADT reads       |

> The following columns, representing QC statistics from [`DropletUtils::cleanTagCounts()`](https://rdrr.io/github/MarioniLab/DropletUtils/man/cleanTagCounts.html), can be further be found in the `"CITEseq"`  alternative experiment `colData`, accessed with `colData(altExp(sce, "CITEseq"))`.

Reviewer: Slight rewording...
```suggestion
In addition, the following QC statistics from [`DropletUtils::cleanTagCounts()`](https://rdrr.io/github/MarioniLab/DropletUtils/man/cleanTagCounts.html) can be found in the `colData` of the `"CITEseq"` alternative experiment, accessed with `colData(altExp(sce, "CITEseq"))`.
```

| Column name                | Contents                                          |
| -------------------------- | ------------------------------------------------- |
| `zero.ambient`   | Indicates whether the cell has zero ambient contamination   |
| `sum.controls` |  The sum of counts for all control features; Only present if negative control ADTs are present. |
| `high.controls`  | Indicates whether the cell has unusually high total control counts; Only present if negative control ADTs are present.|
| `ambient.scale` |  The relative amount of ambient contamination; Only present if negative control ADTs are _not_ present. |
| `high.ambient`  | Indicates whether the cell has unusually high contamination; Only present if negative control ADTs are _not_ present.|

> | `discard`  | Indicates whether the cell should be discarded based on QC statistics; All retained cells in the `_processed.rds` are `FALSE`. |

Reviewer: I don't know if we need the second sentence here; I feel like it may be covered by the discussion at `scpca_filter`, but it doesn't hurt too much.

Author: I didn't delete this, but still can!

> | `zero.ambient`   | Indicates whether the cell has zero ambient contamination   |
> | `sum.controls` |  The sum of counts for all control features; Only present if negative control ADTs are present. |
> | `high.controls`  | Indicates whether the cell has unusually high total control counts; Only present if negative control ADTs are present.|
> | `ambient.scale` |  The relative amount of ambient contamination; Only present if negative control ADTs are _not_ present. |
> | `high.ambient`  | Indicates whether the cell has unusually high contamination; Only present if negative control ADTs are _not_ present.|
> | `discard`  | Indicates whether the cell should be discarded based on QC statistics; All retained cells in the `_processed.rds` are `FALSE`. |

Reviewer: semicolons -> periods
```suggestion
| `zero.ambient`   | Indicates whether the cell has zero ambient contamination   |
| `sum.controls` |  The sum of counts for all control features. Only present if negative control ADTs are present. |
| `high.controls`  | Indicates whether the cell has unusually high total control counts. Only present if negative control ADTs are present.|
| `ambient.scale` |  The relative amount of ambient contamination. Only present if negative control ADTs are _not_ present. |
| `high.ambient`  | Indicates whether the cell has unusually high contamination. Only present if negative control ADTs are _not_ present.|
| `discard`  | Indicates whether the cell should be discarded based on QC statistics. All retained cells in the `_processed.rds` are `FALSE`. |
```

| ----------- | -------------------------------------------------------------- |
| `mean`      | Mean ADT count across all cells/droplets                       |
| `detected`  | Percent of cells in which the ADT was detected (ADT count > 0 ) |
| `target_type` | Whether each ADT is a `target`, `neg_control`, or `pos_control`. This column will be empty if target information is unknown. |

Finally, additional metadata for the CITE-seq data processing can be found in the metadata slot of the alternative experiment.

> This metadata slot has the same contents as the [parent experiment metadata](#experiment-metadata), along with one additional field `ambient_profile` which holds a list of representing the ambient concentrations of each ADT.

Reviewer: ```suggestion
This metadata slot has the same contents as the [parent experiment metadata](#experiment-metadata), along with one additional field, `ambient_profile`, which holds a list of representing the ambient concentrations of each ADT.
```
